src/Medology/Behat/TypeCaster.php

```diff
@@ -10,11 +10,14 @@ trait TypeCaster
     /**
      * Casts a step argument from a string to an int.
      *
-     * @Transform /^(0|[1-9]\d*)$/
+     * Will cast the string to an int if the string is int like and within the max int range of the system.
+     * Otherwise, the original string will be returned unmodified.
+     *
+     * @Transform /^((\-\d+)|\d+)$/
```

Comment:
You have changed this so that it will now match a string that has a leading zero. This should not match that as it will break zip codes that have a leading zero by stripping it out.

You just want to add the optional minus sign before the non-zero repeating digit, because obviously negative zero is not a thing.

```suggestion
     * @Transform /^(0|-?[1-9]\d*)$/
```

Reply:
You had asked about being more productive before, and this was one of the things we went over. When making changes for code review feedback, you should only make the change that was requested. By making an additional change, you've caused yet more code review feedback, slowing down the process.